src/Iris/Colour/Detect.hs

```diff
@@ -0,0 +1,28 @@
+{- |
+Module                  : Iris.Colour.Detect
+Copyright               : (c) 2020 Kowainik
+                          (c) 2022 Dmitrii Kovanikov
+SPDX-License-Identifier : MPL-2.0
+Maintainer              : Dmitrii Kovanikov <kovanikov@gmail.com>
+Stability               : Experimental
+Portability             : Portable
+
+Detects if colour should be enabled or not
+
+@since x.x.x.x
+-}
+
+module Iris.Colour.Detect
+    ( detectColour
+    ) where
+import System.Environment (lookupEnv)
+
+
+
+detectColour :: Maybe String -> IO Bool
+detectColour appname = any  (/= Nothing) <$> traverse lookupEnv varnames
+  where
+    basevarnames = ["NO_COLOR","NO_COLOUR"]
+    varnames = case (<> "_") <$> appname of
+        Nothing -> basevarnames
+        Just n  -> ((n <>) <$> basevarnames) <> basevarnames
```

Comment:
The existing code is a bit hard to read and understand due to the dense usage of operators. I propose to rewrite it to the following, probably a more verbose but more explicit and clear version:

```suggestion
    baseVarNames = ["NO_COLOR", "NO_COLOUR"]
    
    appVarNames = case maybeAppName of
        Nothing -> []
        Just appName -> map (\envName -> appName <> "_" <> envName) baseVarNames

    allVarNames = baseVarNames <> appVarNames
```